src/utils/compute_price_impact.rs

```diff
@@ -14,15 +14,19 @@ pub fn compute_price_impact<TBase: CurrencyTrait, TQuote: CurrencyTrait>(
     mid_price: Price<TBase, TQuote>,
     input_amount: CurrencyAmount<TBase>,
     output_amount: CurrencyAmount<TQuote>,
-) -> Percent {
+) -> Result<Percent, Error> {
     let quoted_output_amount = mid_price.quote(input_amount);
     // calculate price impact := (exactQuote - outputAmount) / exactQuote
-    let price_impact = quoted_output_amount
-        .subtract(&output_amount)
-        .divide(&quoted_output_amount);
+    let price_impact = match quoted_output_amount {
+        Ok(quoted_output_amount) => quoted_output_amount
+            .subtract(&output_amount)?
+            .divide(&quoted_output_amount),
+        Err(e) => Err(e),
+    };
+    let price_impact_clone = price_impact?.clone();
```

Comment:
It doesn't need to be cloned.

Reply:
yeah i know, but i placed it incase price_impact was ever used again in the same closure

Reply:
It won't.